### DISCUSSION TITLE
Building jobrunr 6.0.0 succeeded with JDK17 but unable to locate jobrunr-6.0.0.jar

### DISCUSSION THREAD

**Discussion by nandakishorkn** [jobrunr/jobrunr#731] [General]
Building jobrunr 6.0.0 succeeded with JDK17 but unable to locate jobrunr-6.0.0.jar either under jobrunr source directory or under  local maven repo location.

$ git clone -b v6.0.0 https://github.com/jobrunr/jobrunr.git

$ cd jobrurn

$ java -version
java version "17.0.7" 2023-04-18 LTS
Java(TM) SE Runtime Environment (build 17.0.7+5-LTS-219)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.7+5-LTS-219, mixed mode, sharing)

$ ./gradlew clean build -x test 2>&1 | tee /tmp/jobrunr6jdk17.log
> Task :clean
> Task :core:clean
> Task :framework-support:clean
> Task :language-support:clean
> Task :platform:clean
> Task :tests:clean
> Task :framework-support:jobrunr-micronaut-feature:clean
> Task :framework-support:jobrunr-quarkus-extension:clean
> Task :framework-support:jobrunr-spring-boot-2-starter:clean
> Task :framework-support:jobrunr-spring-boot-3-starter:clean
:
:
:
> Task :framework-support:jobrunr-quarkus-extension:tests:build

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 30s
141 actionable tasks: 103 executed, 38 from cache

I also tried 'publishToMavenLocal' and the build succeeded :
./gradlew publishToMavenLocal 

In both cases, I am unable to locate the jobrunr-6.0.0.jar either under jobrunr source directory or under maven local repo location $HOME/.m2/repository

When I searched for jobrunr*.jar from the jobrunr source code parent directory, I found the following but I did not see jobrunr-6.0.0.jar : 
----------
$ find . -name jobrunr*.jar
./framework-support/jobrunr-micronaut-feature/build/libs/jobrunr-micronaut-feature-1.0.0-SNAPSHOT.jar
./framework-support/jobrunr-micronaut-feature/build/libs/jobrunr-micronaut-feature-1.0.0-SNAPSHOT-javadoc.jar
./framework-support/jobrunr-micronaut-feature/build/libs/jobrunr-micronaut-feature-1.0.0-SNAPSHOT-sources.jar
./framework-support/jobrunr-quarkus-extension/build/libs/jobrunr-quarkus-extension-1.0.0-SNAPSHOT.jar
./framework-support/jobrunr-spring-boot-2-starter/build/libs/jobrunr-spring-boot-2-starter-1.0.0-SNAPSHOT.jar
./framework-support/jobrunr-spring-boot-2-starter/build/libs/jobrunr-spring-boot-2-starter-1.0.0-SNAPSHOT-javadoc.jar
./framework-support/jobrunr-spring-boot-2-starter/build/libs/jobrunr-spring-boot-2-starter-1.0.0-SNAPSHOT-sources.jar
./framework-support/jobrunr-spring-boot-3-starter/build/libs/jobrunr-spring-boot-3-starter-1.0.0-SNAPSHOT.jar
./framework-support/jobrunr-spring-boot-3-starter/build/libs/jobrunr-spring-boot-3-starter-1.0.0-SNAPSHOT-javadoc.jar
./framework-support/jobrunr-spring-boot-3-starter/build/libs/jobrunr-spring-boot-3-starter-1.0.0-SNAPSHOT-sources.jar
./language-support/jobrunr-kotlin-16-support/build/libs/jobrunr-kotlin-16-support-1.0.0-SNAPSHOT.jar
./language-support/jobrunr-kotlin-16-support/build/libs/jobrunr-kotlin-16-support-1.0.0-SNAPSHOT-javadoc.jar
./language-support/jobrunr-kotlin-16-support/build/libs/jobrunr-kotlin-16-support-1.0.0-SNAPSHOT-sources.jar
./language-support/jobrunr-kotlin-17-support/build/libs/jobrunr-kotlin-17-support-1.0.0-SNAPSHOT.jar
./language-support/jobrunr-kotlin-17-support/build/libs/jobrunr-kotlin-17-support-1.0.0-SNAPSHOT-javadoc.jar
./language-support/jobrunr-kotlin-17-support/build/libs/jobrunr-kotlin-17-support-1.0.0-SNAPSHOT-sources.jar
./language-support/jobrunr-kotlin-18-support/build/libs/jobrunr-kotlin-18-support-1.0.0-SNAPSHOT.jar
./language-support/jobrunr-kotlin-18-support/build/libs/jobrunr-kotlin-18-support-1.0.0-SNAPSHOT-javadoc.jar
./language-support/jobrunr-kotlin-18-support/build/libs/jobrunr-kotlin-18-support-1.0.0-SNAPSHOT-sources.jar

But I found core-1.0.0-SNAPSHOT binaries under core/build/libs but size is not matching with jobrunr-6.0.0.jar available on MVN central :
$ ls core/build/libs/
core-1.0.0-SNAPSHOT.jar  core-1.0.0-SNAPSHOT-javadoc.jar  core-1.0.0-SNAPSHOT-sources.jar

I don't see jobrunr 6 jars under local maven repo location too :
$ find $HOME/.m2/repository/ -name jobrunr-6.0.0.jar
$ find $HOME/.m2/repository/ -name jobrunr-6*.jar
$
-----------------

Am I missing something ? Any suggestions ?

[jobrunr6jdk17.log](https://github.com/jobrunr/jobrunr/files/11075032/jobrunr6jdk17.log)

**Comment by rdehuyss** [comment]
They will be named 1.0.0-snapshot in maven local.

Size may not match sure to unpublished changed our because you did not build the frontend as mentioned in the docs.

**Comment by nandakishorkn** [comment]
@rdehuyss If we are not planning to use jobrunr dashboard / frontend, Can we rename core-1.0.0-SNAPSHOT.jar as jobrunr-6.0.0.jar and use it ?

One more thing, though we are building it from v6.0.0 branch, Why is the build system generating core-1.0.0-SNAPSHOT.jar instead of jobrunr-6.0.0.jar ?

How is the artifact ( jobrunr-6.0.0.jar ) uploaded to Maven central ? Where do you pick it to upload to maven central ?

**Comment by nandakishorkn** [comment]
@rdehuyss Look forward for your responses / clarification to my queries ( see my previous reply ).

**Comment by rdehuyss** [comment]
@nandakishorkn it's open-source, so feel free to have a look.
